lib/formats/sh.c.in

```diff
@@ -269,23 +269,23 @@ static void rm_fmt_head(RmSession *session, RmFmtHandler *parent, FILE *out) {
 
     self->line_count = 0;
     self->session = session;
-    self->user_cmd = rm_fmt_get_config_value(session->formats, "sh", "cmd");
+    self->user_cmd = rm_fmt_get_config_value(session->cfg->formats, "sh", "cmd");
```

Comment:
Actually I think I'm using the wrong terminology.  [These variables](https://github.com/SeeSpotRun/rmlint/blob/e6fe256470d6202f1b1524ea7361a61749944f21/lib/formats.c#L34-L63) are not "globals", they are "variables with file scope".  And they are encapsulated.
> would not work anymore (or just once...)

Exactly.  For things where "just once" makes sense (e.g. [a thread pool](https://github.com/GNOME/glib/blob/master/glib/gthreadpool.c#L109-L113)), encapsulated variables with file scope make sense.  It probably makes sense for our [RmMountTable](https://github.com/sahib/rmlint/blob/master/lib/utilities.h#L325-L390) (there is only ever one set of mounts).
> I'm not willing to go there

I confess my initial reaction was to continue the debate.  I was going to say it's impossible to think of any case where someone would want two separate RmFmtTable instance.  But maybe it's not, e.g.:
```
session->big_file_outputs = rm_fmt_tables_new();  // files >1M get sent to these formatters
session->small_file_outputs = rm_fmt_tables_new(); // little files get sent to these formatters
```
Not a particularly convincing example but enough to convince myself.  So let me reinstate session->formats into https://github.com/SeeSpotRun/rmlint/tree/formats-refactoring and see how that looks.